### PR TITLE
BusinessLineReporter: Partially eager-load queries; add basic test

### PR DESCRIPTION
**Pull request by n1zyy**
### Description
I noticed that DecisionReviewsController#index was pretty slow in New Relic: https://onenr.io/0YBR635N1jO

While the slowest component was BGS calls, it also showed an average of over 4,000 "Postgres RequestIssue find" calls. This PR _partially_ improves that, but there are still some one-off queries that are less easily fixed. The focus of this innovation project was to find easy wins, so I'd like to call this PR complete: it reduces the number of queries triggered from within a loop.

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] Code appears safe

### Testing Plan
No behavior change. Also, I added a test for something that previously had none. 😛

### CODE REVIEW

File: app/services/business_line_reporter.rb [hunk 1]
```diff
@@ -12,7 +12,9 @@ def initialize(business_line)
   end
 
   def tasks
-    business_line.tasks.completed.includes([:assigned_to, :appeal]).order(id: :asc)
+    business_line.tasks.completed.includes(
+      [:assigned_to, appeal: [:request_issues, :decision_issues, intake: [:user]]]
+    ).order(id: :asc)
```
Comment (n1zyy) [inline app/services/business_line_reporter.rb:17]: Including request_issues and decision_issues through the appeal association is key to eager-loading request issues and decision issues. Also brought in intake and its user association that way.

File: app/services/business_line_reporter.rb [hunk 2]
```diff
@@ -26,8 +28,8 @@ def as_csv
           task.appeal_id,
           task.appeal_type,
           task.appeal.claimant&.name,
-          task.appeal.request_issues&.count,
-          task.appeal.decision_issues&.count,
+          task.appeal.request_issues.size,
+          task.appeal.decision_issues.size,
```
Comment (n1zyy) [inline app/services/business_line_reporter.rb:32]: TIL: `.count` appears to always cause ActiveRecord to do a `SELECT COUNT` query, even if you eager-load the association. Making this `size` instead of `count` causes it to look at array length instead.

I need to change these to `&.size` because `nil.size` is still invalid. I briefly had `to_a.size` to coerce it, but this turned out to be unnecessary.

File: spec/services/business_line_reporter_spec.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+describe 'BusinessLineReporter' do
+  let(:business_line) { create(:business_line) }
+
+  before do
+    10.times do
+      appeal = create(:appeal, :with_post_intake_tasks)
+      ama_task = create(:ama_task, appeal: appeal, assigned_to: business_line)
+      ama_task.completed!
+    end
```
Comment (n1zyy) [inline spec/services/business_line_reporter_spec.rb:11]: Honestly don't even review this test right now. It's currently just to set up the one-off queries.

BusinessLineReporter appears to have had no test. 🤨

Comment (n1zyy) [reply spec/services/business_line_reporter_spec.rb:11]: Without the changes to BusinessLineReporter:

We do two `select count` queries, one for decision issues and one for request issues. This is repeated 10 times because I set up 10 appeals.

```sql
[2021-09-07 20:31:06 -0400]   Task Load (1.8ms)  SELECT "tasks".* FROM "tasks" WHERE "tasks"."assigned_to_id" = $1 AND "tasks"."assigned_to_type" = $2 AND "tasks"."status" = $3 ORDER BY "tasks"."id" ASC  [["assigned_to_id", 3], ["assigned_to_type", "Organization"], ["status", "completed"]]
[2021-09-07 20:31:06 -0400]   Appeal Load (1.9ms)  SELECT "appeals".* FROM "appeals" WHERE "appeals"."id" IN ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10)  [["id", 1], ["id", 2], ["id", 3], ["id", 4], ["id", 5], ["id", 6], ["id", 7], ["id", 8], ["id", 9], ["id", 10]]
[2021-09-07 20:31:06 -0400]   Claimant Load (1.5ms)  SELECT  "claimants".* FROM "claimants" WHERE "claimants"."decision_review_id" = $1 AND "claimants"."decision_review_type" = $2 ORDER BY "claimants"."id" DESC LIMIT $3  [["decision_review_id", 1], ["decision_review_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Person Load (1.2ms)  SELECT  "people".* FROM "people" WHERE "people"."participant_id" = $1 LIMIT $2  [["participant_id", "500000000"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]    (1.5ms)  SELECT COUNT(*) FROM "request_issues" WHERE "request_issues"."decision_review_id" = $1 AND "request_issues"."decision_review_type" = $2  [["decision_review_id", 1], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]    (1.4ms)  SELECT COUNT(*) FROM "decision_issues" WHERE "decision_issues"."deleted_at" IS NULL AND "decision_issues"."decision_review_id" = $1 AND "decision_issues"."decision_review_type" = $2  [["decision_review_id", 1], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]   Intake Load (1.4ms)  SELECT  "intakes".* FROM "intakes" WHERE "intakes"."detail_id" = $1 AND "intakes"."detail_type" = $2 LIMIT $3  [["detail_id", 1], ["detail_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Claimant Load (1.0ms)  SELECT  "claimants".* FROM "claimants" WHERE "claimants"."decision_review_id" = $1 AND "claimants"."decision_review_type" = $2 ORDER BY "claimants"."id" DESC LIMIT $3  [["decision_review_id", 2], ["decision_review_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Person Load (1.0ms)  SELECT  "people".* FROM "people" WHERE "people"."participant_id" = $1 LIMIT $2  [["participant_id", "500000001"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]    (1.1ms)  SELECT COUNT(*) FROM "request_issues" WHERE "request_issues"."decision_review_id" = $1 AND "request_issues"."decision_review_type" = $2  [["decision_review_id", 2], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]    (1.3ms)  SELECT COUNT(*) FROM "decision_issues" WHERE "decision_issues"."deleted_at" IS NULL AND "decision_issues"."decision_review_id" = $1 AND "decision_issues"."decision_review_type" = $2  [["decision_review_id", 2], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]   Intake Load (1.1ms)  SELECT  "intakes".* FROM "intakes" WHERE "intakes"."detail_id" = $1 AND "intakes"."detail_type" = $2 LIMIT $3  [["detail_id", 2], ["detail_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Claimant Load (1.2ms)  SELECT  "claimants".* FROM "claimants" WHERE "claimants"."decision_review_id" = $1 AND "claimants"."decision_review_type" = $2 ORDER BY "claimants"."id" DESC LIMIT $3  [["decision_review_id", 3], ["decision_review_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Person Load (1.2ms)  SELECT  "people".* FROM "people" WHERE "people"."participant_id" = $1 LIMIT $2  [["participant_id", "500000002"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]    (1.3ms)  SELECT COUNT(*) FROM "request_issues" WHERE "request_issues"."decision_review_id" = $1 AND "request_issues"."decision_review_type" = $2  [["decision_review_id", 3], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]    (1.3ms)  SELECT COUNT(*) FROM "decision_issues" WHERE "decision_issues"."deleted_at" IS NULL AND "decision_issues"."decision_review_id" = $1 AND "decision_issues"."decision_review_type" = $2  [["decision_review_id", 3], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]   Intake Load (1.8ms)  SELECT  "intakes".* FROM "intakes" WHERE "intakes"."detail_id" = $1 AND "intakes"."detail_type" = $2 LIMIT $3  [["detail_id", 3], ["detail_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Claimant Load (1.5ms)  SELECT  "claimants".* FROM "claimants" WHERE "claimants"."decision_review_id" = $1 AND "claimants"."decision_review_type" = $2 ORDER BY "claimants"."id" DESC LIMIT $3  [["decision_review_id", 4], ["decision_review_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Person Load (1.1ms)  SELECT  "people".* FROM "people" WHERE "people"."participant_id" = $1 LIMIT $2  [["participant_id", "500000003"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]    (1.3ms)  SELECT COUNT(*) FROM "request_issues" WHERE "request_issues"."decision_review_id" = $1 AND "request_issues"."decision_review_type" = $2  [["decision_review_id", 4], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]    (1.4ms)  SELECT COUNT(*) FROM "decision_issues" WHERE "decision_issues"."deleted_at" IS NULL AND "decision_issues"."decision_review_id" = $1 AND "decision_issues"."decision_review_type" = $2  [["decision_review_id", 4], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]   Intake Load (1.2ms)  SELECT  "intakes".* FROM "intakes" WHERE "intakes"."detail_id" = $1 AND "intakes"."detail_type" = $2 LIMIT $3  [["detail_id", 4], ["detail_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Claimant Load (1.2ms)  SELECT  "claimants".* FROM "claimants" WHERE "claimants"."decision_review_id" = $1 AND "claimants"."decision_review_type" = $2 ORDER BY "claimants"."id" DESC LIMIT $3  [["decision_review_id", 5], ["decision_review_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Person Load (1.3ms)  SELECT  "people".* FROM "people" WHERE "people"."participant_id" = $1 LIMIT $2  [["participant_id", "500000004"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]    (1.3ms)  SELECT COUNT(*) FROM "request_issues" WHERE "request_issues"."decision_review_id" = $1 AND "request_issues"."decision_review_type" = $2  [["decision_review_id", 5], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]    (1.3ms)  SELECT COUNT(*) FROM "decision_issues" WHERE "decision_issues"."deleted_at" IS NULL AND "decision_issues"."decision_review_id" = $1 AND "decision_issues"."decision_review_type" = $2  [["decision_review_id", 5], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]   Intake Load (1.2ms)  SELECT  "intakes".* FROM "intakes" WHERE "intakes"."detail_id" = $1 AND "intakes"."detail_type" = $2 LIMIT $3  [["detail_id", 5], ["detail_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Claimant Load (1.2ms)  SELECT  "claimants".* FROM "claimants" WHERE "claimants"."decision_review_id" = $1 AND "claimants"."decision_review_type" = $2 ORDER BY "claimants"."id" DESC LIMIT $3  [["decision_review_id", 6], ["decision_review_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Person Load (1.3ms)  SELECT  "people".* FROM "people" WHERE "people"."participant_id" = $1 LIMIT $2  [["participant_id", "500000005"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]    (1.4ms)  SELECT COUNT(*) FROM "request_issues" WHERE "request_issues"."decision_review_id" = $1 AND "request_issues"."decision_review_type" = $2  [["decision_review_id", 6], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]    (1.3ms)  SELECT COUNT(*) FROM "decision_issues" WHERE "decision_issues"."deleted_at" IS NULL AND "decision_issues"."decision_review_id" = $1 AND "decision_issues"."decision_review_type" = $2  [["decision_review_id", 6], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]   Intake Load (1.2ms)  SELECT  "intakes".* FROM "intakes" WHERE "intakes"."detail_id" = $1 AND "intakes"."detail_type" = $2 LIMIT $3  [["detail_id", 6], ["detail_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Claimant Load (1.3ms)  SELECT  "claimants".* FROM "claimants" WHERE "claimants"."decision_review_id" = $1 AND "claimants"."decision_review_type" = $2 ORDER BY "claimants"."id" DESC LIMIT $3  [["decision_review_id", 7], ["decision_review_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Person Load (1.3ms)  SELECT  "people".* FROM "people" WHERE "people"."participant_id" = $1 LIMIT $2  [["participant_id", "500000006"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]    (1.4ms)  SELECT COUNT(*) FROM "request_issues" WHERE "request_issues"."decision_review_id" = $1 AND "request_issues"."decision_review_type" = $2  [["decision_review_id", 7], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]    (1.6ms)  SELECT COUNT(*) FROM "decision_issues" WHERE "decision_issues"."deleted_at" IS NULL AND "decision_issues"."decision_review_id" = $1 AND "decision_issues"."decision_review_type" = $2  [["decision_review_id", 7], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]   Intake Load (1.2ms)  SELECT  "intakes".* FROM "intakes" WHERE "intakes"."detail_id" = $1 AND "intakes"."detail_type" = $2 LIMIT $3  [["detail_id", 7], ["detail_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Claimant Load (1.2ms)  SELECT  "claimants".* FROM "claimants" WHERE "claimants"."decision_review_id" = $1 AND "claimants"."decision_review_type" = $2 ORDER BY "claimants"."id" DESC LIMIT $3  [["decision_review_id", 8], ["decision_review_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Person Load (1.2ms)  SELECT  "people".* FROM "people" WHERE "people"."participant_id" = $1 LIMIT $2  [["participant_id", "500000007"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]    (1.4ms)  SELECT COUNT(*) FROM "request_issues" WHERE "request_issues"."decision_review_id" = $1 AND "request_issues"."decision_review_type" = $2  [["decision_review_id", 8], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]    (1.4ms)  SELECT COUNT(*) FROM "decision_issues" WHERE "decision_issues"."deleted_at" IS NULL AND "decision_issues"."decision_review_id" = $1 AND "decision_issues"."decision_review_type" = $2  [["decision_review_id", 8], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]   Intake Load (1.2ms)  SELECT  "intakes".* FROM "intakes" WHERE "intakes"."detail_id" = $1 AND "intakes"."detail_type" = $2 LIMIT $3  [["detail_id", 8], ["detail_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Claimant Load (1.2ms)  SELECT  "claimants".* FROM "claimants" WHERE "claimants"."decision_review_id" = $1 AND "claimants"."decision_review_type" = $2 ORDER BY "claimants"."id" DESC LIMIT $3  [["decision_review_id", 9], ["decision_review_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Person Load (1.2ms)  SELECT  "people".* FROM "people" WHERE "people"."participant_id" = $1 LIMIT $2  [["participant_id", "500000008"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]    (1.3ms)  SELECT COUNT(*) FROM "request_issues" WHERE "request_issues"."decision_review_id" = $1 AND "request_issues"."decision_review_type" = $2  [["decision_review_id", 9], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]    (1.3ms)  SELECT COUNT(*) FROM "decision_issues" WHERE "decision_issues"."deleted_at" IS NULL AND "decision_issues"."decision_review_id" = $1 AND "decision_issues"."decision_review_type" = $2  [["decision_review_id", 9], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]   Intake Load (1.2ms)  SELECT  "intakes".* FROM "intakes" WHERE "intakes"."detail_id" = $1 AND "intakes"."detail_type" = $2 LIMIT $3  [["detail_id", 9], ["detail_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Claimant Load (1.3ms)  SELECT  "claimants".* FROM "claimants" WHERE "claimants"."decision_review_id" = $1 AND "claimants"."decision_review_type" = $2 ORDER BY "claimants"."id" DESC LIMIT $3  [["decision_review_id", 10], ["decision_review_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]   Person Load (1.3ms)  SELECT  "people".* FROM "people" WHERE "people"."participant_id" = $1 LIMIT $2  [["participant_id", "500000009"], ["LIMIT", 1]]
[2021-09-07 20:31:06 -0400]    (1.2ms)  SELECT COUNT(*) FROM "request_issues" WHERE "request_issues"."decision_review_id" = $1 AND "request_issues"."decision_review_type" = $2  [["decision_review_id", 10], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]    (1.3ms)  SELECT COUNT(*) FROM "decision_issues" WHERE "decision_issues"."deleted_at" IS NULL AND "decision_issues"."decision_review_id" = $1 AND "decision_issues"."decision_review_type" = $2  [["decision_review_id", 10], ["decision_review_type", "Appeal"]]
[2021-09-07 20:31:06 -0400]   Intake Load (1.3ms)  SELECT  "intakes".* FROM "intakes" WHERE "intakes"."detail_id" = $1 AND "intakes"."detail_type" = $2 LIMIT $3  [["detail_id", 10], ["detail_type", "Appeal"], ["LIMIT", 1]]
[2021-09-07 20:31:07 -0400]    (1.4ms)  ROLLBACK
```

**Review by Jeremy1026 — APPROVED** [review]
LGTM

File: app/services/business_line_reporter.rb [hunk 4]
```diff
@@ -26,8 +28,8 @@ def as_csv
           task.appeal_id,
           task.appeal_type,
           task.appeal.claimant&.name,
```
[the remark this answers is not in the document]
Comment (nanotone) [reply app/services/business_line_reporter.rb:30]: Definitely don't optimize `claimants` because it's possible we need to re-think that whole thing in light up upcoming Contested Claims work.

**Review by nanotone — APPROVED** [review]
LGTM
